libhdf5/hdf5create.c

```diff
@@ -192,10 +192,15 @@ nc4_create_file(const char *path, int cmode, size_t initialsz,
     if (H5Pset_link_creation_order(fcpl_id, (H5P_CRT_ORDER_TRACKED |
                                              H5P_CRT_ORDER_INDEXED)) < 0)
         BAIL(NC_EHDFERR);
-    if (H5Pset_attr_creation_order(fcpl_id, (H5P_CRT_ORDER_TRACKED |
-                                             H5P_CRT_ORDER_INDEXED)) < 0)
-        BAIL(NC_EHDFERR);
 
+    if (cmode & NC_NOATTCREORD) {
```

Comment:
I think we need to make it persist. Suppose we create a file without creation order.
Then close it. Then later open it and add more attributes with creation order enabled.
What should happen? BTW is there a test in HDF5 to inquire if creation order is set?

Reply:
There is a test in HDF5 for creation order, IIRC.

Reply:
There is `herr_t H5Pget_attr_creation_order( hid_t ocpl_id, unsigned *crt_order_flags )`.  Since HDF5 allows setting this on an object by object (variable in netCDF case) basis, we could query it at metadata read time after the open and then set the `no_attr_create_order` flag if all(?) or any(?) of the variables have the attribute creation order disabled?

Reply:
OK, latest commit detects the attribute creation order setting on the input file and if opened with NC_WRITE, will persist that setting for subsequent variable / attribute creation.